Source/WebCore/platform/cocoa/ParentalControlsContentFilter.mm

```diff
@@ -184,15 +187,10 @@ static inline bool canHandleResponse(const ResourceResponse& response)
 
 #if HAVE(WEBCONTENTRESTRICTIONS)
 
-void ParentalControlsContentFilter::updateFilterState(bool shouldBlock, NSData *replacementData)
+void ParentalControlsContentFilter::updateFilterState(bool shouldBlock, Vector<uint8_t>&& replacementData)
```

Comment:
Looks like we just need a span here, not a Vector.

Reply:
I'm actually not entirely clear why you went from a NSData to a Vector. It seems we get a NSData from the underlying framework and store it as an NSData in the end. It seems a bit wasteful. Switching from Vector to span would make 1 copy of the bytes instead of 2. But it looks like if we kept using `NSData` all the way through, there would be 0 copies?

Reply:
Changed `CompletionHandler` to take `NSData*` instead.